src/iterators/linear.rs

```diff
@@ -48,6 +48,7 @@ impl<'a, T: 'a + Counter> PickyIterator<T> for Iter<'a, T> {
         // if we reached this point), then we are not yet done iterating (we want to iterate
         // until we are no longer on a value that has a count, rather than util we first reach
         // the last value that has a count. The difference is subtle but important)...
+        // TODO index + 1 could overflow 16-bit usize
```

Comment:
Sometimes I wonder if AVR support is worth the pain... but this would be such a great data structure for sensor data